src/libsystemd/sd-journal/test-journal-flush.c

```diff
@@ -17,6 +19,83 @@
 #include "tests.h"
 #include "tmpfile-util.h"
 
+static int open_archive_file(sd_journal **ret) {
+        _cleanup_closedir_ DIR *d = NULL;
+        _cleanup_close_ int newest_fd = -EBADF;
+        unsigned long long newest_realtime = 0;
+        bool newest_is_system = false;
+        sd_id128_t machine_id;
+        const char *p;
+        int r;
+
+        r = sd_id128_get_machine(&machine_id);
+        if (r < 0)
+                return r;
+
+        p = strjoina("/var/log/journal/", SD_ID128_TO_STRING(machine_id), "/");
```

Comment:
Hmm this assumes we have a persistent journal. I'm considering making mkosi CI tests use only a runtime journal and not a persistent one, but this test would break then.

Reply:
If there is no persistent journal, the test falls back to use runtime journal.